hooks/shapediver/viewer/interaction/useSelectManagerEvents.ts

```diff
@@ -0,0 +1,133 @@
+import { notifications } from "@mantine/notifications";
+import { addListener, EVENTTYPE_INTERACTION, IEvent, removeListener } from "@shapediver/viewer";
+import { InteractionEventResponseMapping, MultiSelectManager } from "@shapediver/viewer.features.interaction";
+import { useState, useCallback, useEffect } from "react";
+import { NameFilterPattern, processNodes } from "./utils/patternUtils";
+
+// #region Functions (1)
+
+/**
+ * Hook allowing to create the select manager events.
+ * 
+ * @param pattern The pattern to match the hovered nodes.
+ */
+export function useSelectManagerEvents(pattern: NameFilterPattern): {
+	/**
+	 * The selected node names.
+	 */
+    selectedNodeNames: string[],
+	/**
+	 * Callback function to reset the selected node names.
+	 * 
+	 * @returns 
+	 */
+    resetSelectedNodeNames: () => void
+} {
+	// state for the selected nodes
+	const [selectedNodeNames, setSelectedNodeNames] = useState<string[]>([]);
+	const resetSelectedNodeNames = useCallback(() => setSelectedNodeNames([]), []);
+
+	// register an event handler and listen for output updates
+	useEffect(() => {
+		/**
+         * Event handler for the select on event.
+         * In this event handler, the response object is created and the parameter is updated.
```

Comment:
@MajorMeerkatThe3rd  Cosmetics: This comment is a leftover (appears similar in other places)